packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnGrouping.ts

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { GridApiCommunity } from '../../../models/api/gridApiCommunity';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import {
+  GridColumnGroupingModel,
+  GridColumnNode,
+  isLeaf,
+} from '../../../models/gridColumnGrouping';
+import { gridColumnGroupsLookupSelector, gridColumnLookupSelector } from './gridColumnsSelector';
+import { GridColumnGroupLookup } from './gridColumnsInterfaces';
+import { GridColumnGroupingApi } from '../../../models/api/gridColumnGroupingApi';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridStateColDef, GridColDef } from '../../../models/colDef';
+
+export function hasGroupPath(
+  lookupElement: GridColDef | GridStateColDef,
+): lookupElement is GridStateColDef {
+  return (<GridStateColDef>lookupElement).groupPath !== undefined;
+}
+
+type UnwrappedGroupingModel = { [key: string]: string[] };
+
+const recurrentUnwrapGroupingColumnModel = (
+  columnGroupNode: GridColumnNode,
+  parents: any,
+): UnwrappedGroupingModel => {
+  if (isLeaf(columnGroupNode)) {
+    return { [columnGroupNode.field]: parents };
+  }
+  const rep: UnwrappedGroupingModel = {};
+  const { groupId, children } = columnGroupNode;
+  children.forEach((child) => {
+    const unwrappedSubTree = recurrentUnwrapGroupingColumnModel(child, [...parents, groupId]);
+    Object.entries(unwrappedSubTree).forEach(([key, value]) => {
+      if (rep[key] !== undefined) {
+        throw new Error(
+          [
+            `MUI DataGrid - column grouping: duplicated field`,
```

Comment:
```suggestion
            `MUI: Column grouping: duplicated field`,
```


If we want to change the prefix we should do it everwhere (in this PR you use `MUI`, `MUI DataGrid` and `MUI-DataGrid`)